src/Language/PureScript/Pretty/JS.hs

```diff
@@ -162,8 +162,10 @@ conditional = mkPattern match
 accessor :: (Emit gen) => Pattern PrinterState JS (gen, JS)
 accessor = mkPattern match
   where
-  -- WARN: if `prop` does not match the `IdentifierName` grammar, this will generate invalid code; see #2513
-  match (JSAccessor _ prop val) = Just (emit prop, val)
+  match (JSIndexer _ (JSStringLiteral _ prop) val) =
```

Comment:
I don't like the idea of pattern synonyms. I think the additional indirection will be confusing. I think this is the only place it'd be used anyway.